src/Aspire.Hosting/Utils/HostNameResolver.cs

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Configuration;
+
+namespace Aspire.Hosting.Utils;
+
+internal class HostNameResolver
+{
+    // HACK: When the destination resource is a container, we need to replace the localhost with host.docker.internal
+    // so the container can access the other container via the host's IP address.
+    internal static string ReplaceLocalhostWithContainerHost(string value, IConfiguration configuration)
+    {
+        // https://stackoverflow.com/a/43541732/45091
+
+        // This configuration value is a workaround for the fact that host.docker.internal is not available on Linux by default.
+        var hostName = configuration["AppHost:ContainerHostname"] ?? "host.docker.internal";
+
+        // REVIEW: Handle IPV6?
+        return value.Replace("localhost", hostName, StringComparison.OrdinalIgnoreCase)
+                    .Replace("127.0.0.1", hostName);
```

Comment:
Aspire uses `localhost` by default, but the AppHost code is free to change that. I would add the IPV6 `[::1]` to the replacement list. 

Other IPv4 loopback addresses (not `127.0.0.1`) will not work, but they are probably very unlikely to be used, and we are going to rework this in next release.